### PR TITLE
Add doc for the new package_version comparisons

### PR DESCRIPTION
**Pull request by rtorrero**
This PR documents the changes introduced in PR https://github.com/trento-project/agent/pull/169

### CODE REVIEW

**Review by fabriziosestito — CHANGES_REQUESTED** [review]
@rtorrero let's wait to merge #160 and base this PR on the new format

**Review by nelsonkopliku — COMMENTED** [review]
Good job!

When it comes to the spec, we might want to consider suggesting putting some emphasis in facts and expectation names to clarify further. 

```yaml
expect: facts.package_corosync = 1 # or  facts.package_corosync > 0
expect: facts.package_corosync = 0
expect: facts.package_corosync = -1 # or facts.package_corosync < 0
```
At a first glance `facts.package_corosync = 1` might not sound immediately telling _Installed Corosync version must be equal or higher than the recommended value_ (DC5429)

File: guides/gatherers.md [hunk 1]
```diff
@@ -269,13 +269,21 @@ Example output (in Rhai):
...
 
-This gatherer returns the version as a string of the specified rpm package.
+This gatherer supports two usecases:
+ - it can be used to return the version as a string of the specified package.
+ - it can be used to compare a given version string against the installed version string of a given package.
+
+While for the first usecase, a simple string containing the version is returned, for the second usecase, the return value is as follows:
+ - A value of `0` if the provided version string matches the installed package version for the requested package.
+ - A value of `-1` if the provided version string is older that what's currently installed.
+ - A value of `1` if the provided version string is newer than what's currently installed.
```
Comment (fabriziosestito) [inline guides/gatherers.md:279]: we can mention the rpmvercmp algorithm somewhere here (with a link to the algorithm i.e. https://fedoraproject.org/wiki/Archive:Tools/RPM/VersionComparison#The_rpmvercmp_algorithm) as the user might be familiar with it

**Review by fabriziosestito — APPROVED** [review]
Just one comment then LGTM :shipit:

**Review by nelsonkopliku — APPROVED** [review]

**Pull request merged**